client/src/components/generic/TextInputComponent/TextInput.tsx

```diff
@@ -1,10 +1,23 @@
-import React from "react"
+import React, { ChangeEventHandler } from "react"
 
 type TextInputProps = React.InputHTMLAttributes<HTMLInputElement> & {
   variant?: "success" | "error"
+  label?: string
+  description?: string
+  id?: string
+  value: string | undefined
+  onChange: ChangeEventHandler<HTMLInputElement>
 }
 
-const TextInput = ({ variant, ...props }: TextInputProps) => {
+const TextInput = ({
+  variant,
+  label,
+  description,
+  id,
+  value,
```

Comment:
we dont need to specify these because they are already included in `React.InputHTMLAttributes<HTMLInputElement>` and they are spread on the input using `{...props}`

Reply:
`value`, `id`, `onChange`